ddtrace/tracer/transport.go

```diff
@@ -127,10 +127,23 @@ func (t *httpTransport) sendStats(p *pb.ClientStatsPayload, tracerObfuscationVer
 	if err := msgp.Encode(&buf, p); err != nil {
 		return err
 	}
-	req, err := http.NewRequest("POST", t.statsURL, &buf)
+	body := buf.Bytes()
+
+	req, err := http.NewRequest("POST", t.statsURL, bytes.NewReader(body))
 	if err != nil {
 		return err
 	}
+
+	// by providing GetBody and a zero length slice for the Idempotency-Key
```

Comment:
This seems like it could work well here, but wouldn't this mean all these requests are considered idempotent so even on a network error where the agent received and processed the request this would cause a retry of the same request? I took a look at the net/http docs for clarity here but they aren't clarifying this exact case for me